api/sip/client.py

```diff
@@ -417,9 +423,30 @@ def make_secure_connection(self):
             fd, tmp_ssl_key_path = tempfile.mkstemp()
             os.write(fd, self.ssl_key.encode("utf-8"))
             os.close(fd)
-        connection = self.make_insecure_connection()
-        connection = ssl.wrap_socket(
-            connection, certfile=tmp_ssl_cert_path, keyfile=tmp_ssl_key_path
+        insecure_connection = self.make_insecure_connection()
+
+        context = self.ssl_contexts(ssl.PROTOCOL_TLS_CLIENT)
+        context.minimum_version = ssl.TLSVersion.TLSv1_2
+        context.load_default_certs(purpose=ssl.Purpose.SERVER_AUTH)
+        context.load_verify_locations(cafile=certifi.where())
+
+        # If the certificate path is provided, the certificate will be loaded.
+        # The private key is dependent on the certificate, so can't be loaded if
+        # there's no certificate present.
+        if tmp_ssl_cert_path:
+            context.load_cert_chain(
+                certfile=tmp_ssl_cert_path, keyfile=tmp_ssl_key_path
+            )
+
+        if self.ssl_no_verification:
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
+        else:
+            context.check_hostname = True
+            context.verify_mode = ssl.CERT_REQUIRED
```

Comment:
This is minor, but do we need the `else` side of the conditional here? I think `self.ssl_contexts(ssl.PROTOCOL_TLS_CLIENT)` sets these values.

Reply:
`ssl.py` is such a mess of poorly specified behaviour, changing defaults, and so on, that I felt it better to be explicit.